### ISSUE TITLE
StoreModel::Model#== cannot properly compare enums

### ISSUE THREAD

**Issue by FunkyloverOne** [DmitryTsepelev/store_model#28]
[Here:](https://github.com/DmitryTsepelev/store_model/blob/master/lib/store_model/model.rb#L38)
```ruby
    def ==(other)
      return super unless other.is_a?(self.class)

      attributes.all? { |name, value| value == other.send(name) }
    end
```

The `value` in this case is going to be an integer, while the result of `other.send(name)` will be a string enum key.

**Comment by DmitryTsepelev** [comment]
Hi @FunkyloverOne, nice catch! Could you please take a look at my [PR](https://github.com/DmitryTsepelev/store_model/pull/29) with the fix before I merge it?

**Comment by DmitryTsepelev** [comment]
Released in `0.5.2`
